src/desktop/apps/article/ad_category.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Parses article metadata to determine correct category type for Google DFP ad placements.
+ */
+
+interface ArticleProperties {
+  layout?: string
+  tracking_tags?: [string]
+  sponsor?: {
+    pixel_tracking_code?: string
+    partner_logo_link?: string
+    partner_light_logo?: string
+    partner_dark_logo?: string
+    partner_condensed_logo?: string
+  }
+}
+
+enum AdCategory {
+  Feature = "feature",
+  Article = "article",
+  NewsLanding = "newslanding",
+  SponsorLanding = "sponsorlanding",
+  SponsorFeature = "sponsorfeature",
+}
+
+const isSponsored = (article: ArticleProperties) => {
+  const { tracking_tags, sponsor } = article
+  if (tracking_tags.includes("sponsored")) {
+    return true
+  }
+
+  // tslint:disable:no-extra-boolean-cast
+  // Check each of the properties of article.sponsor, if any are true then the article is sponsored.
+  if (!!Reflect.get(sponsor, "partner_condensed_logo")) {
+    return true
+  }
+  if (!!Reflect.get(sponsor, "partner_dark_logo")) {
+    return true
+  }
+  if (!!Reflect.get(sponsor, "partner_light_logo")) {
+    return true
+  }
+  if (!!Reflect.get(sponsor, "partner_logo_link")) {
```

Comment:
It would be simpler to to use `some` and then just touch the property directly (no need for `Reflect`):

```js
const isSponsored = ['foo', 'bar', 'baz'].some(prop => sponsor[pro] != null) // != null === not null or undefined
```